crates/bootstrap_srv/src/server.rs

```diff
@@ -0,0 +1,333 @@
+//! bootstrap http server types.
+
+use std::sync::Arc;
+
+use crate::*;
+use tiny_http::*;
+
+/// An actual kitsune2_bootstrap_srv server instance.
+///
+/// This server is built to be direct, light-weight, and responsive.
+/// On the server-side, as one aspect toward accomplishing this,
+/// we are eschewing async code in favor of os thread workers.
+pub struct BootstrapSrv {
+    cont: Arc<std::sync::atomic::AtomicBool>,
+    workers: Vec<std::thread::JoinHandle<std::io::Result<()>>>,
+    addr: std::net::SocketAddr,
+}
+
+impl Drop for BootstrapSrv {
+    fn drop(&mut self) {
+        self.cont.store(false, std::sync::atomic::Ordering::SeqCst);
+        for worker in self.workers.drain(..) {
+            let _ = worker.join().expect("Failure shutting down worker thread");
```

Comment:
Is there any value in passing the error through here?

Reply:
What would you like to pass this through? This is a drop impl, we don't have many options. Technically you're not supposed to panic in drop impls, you'll get a stack dump that is not very useful... I'm open to other suggestions... but rust's handling of panics in other threads is pretty poor, I haven't really heard of a better pattern.

Reply:
It needn't be the main thread if the server were passed to a different thread. But yes, whichever thread the server instance variable is dropped on will panic.

Reply:
Won't that stop the whole server when it's only about dropping workers?

Reply:
I'd suggest that this doesn't have to be a drop implementation here. The `main` program is doing the drop explicitly but you only need to create one instance of this per program, so it could equally be a `shutdown` method that can return an error?

Reply:
A drop implementation that calls `shutdown().unwrap()` would be good, to make sure that the user gets some attempt at shutdown if they forget to call it

Reply:
Done: https://github.com/holochain/kitsune2/pull/10/commits/ebac6ff5dcaff4f2b78c66fab4e696e75c504847